src/browser/disabled_frame_group_tracker.ts

```diff
@@ -92,89 +101,110 @@ function handleApiMove(win: OpenFinWindow, delta: RectangleBase) {
+        return accum;
+    }, <MoveAccumulator>{otherWindows: []});
+    if (!leader || leader[1].moved(newBounds)) {
         //Propsed move differs from requested move
         throw new Error('Attempted move violates group constraints');
     }
     handleBatchedMove(moves);
     emitChange(leader, changeType, 'self', 'changed');
-    const otherWindows = moves.filter(([w]) => w !== win);
+    // const otherWindows = moves.filter(([w]) => w !== win);
     otherWindows.forEach(move => emitChange(move, changeType, 'group', 'changed'));
-    return leader[1].eventBounds;
+    return leader[1];
 }
 
-function handleBatchedMove(moves: [OpenFinWindow, Rectangle][], bringWinsToFront: boolean = false) {
+function handleBatchedMove(moves: Move[], bringWinsToFront: boolean = false) {
     if (isWin32) {
         const { flag: { noZorder, noSize, noActivate } } = WindowTransaction;
         const flags = noZorder + noActivate;
         const wt = new WindowTransaction.Transaction(0);
-        moves.forEach(([win, rect]) => {
+        moves.forEach(([win, rect, offset]) => {
             const hwnd = parseInt(win.browserWindow.nativeId, 16);
-            wt.setWindowPos(hwnd, { ...rect.transactionBounds, flags });
+            wt.setWindowPos(hwnd, { ...getTransactionBounds(rect, offset), flags });
             if (bringWinsToFront) { win.browserWindow.bringToFront(); }
         });
         wt.commit();
     } else {
-        moves.forEach(([win, rect]) => {
-            win.browserWindow.setBounds(rect.bounds);
+        moves.forEach(([win, rect, offset]) => {
+            win.browserWindow.setBounds(applyOffset(rect, offset));
             if (bringWinsToFront) { win.browserWindow.bringToFront(); }
         });
     }
 }
-const makeTranslate = (delta: RectangleBase) => ([win, rect]: Move): Move => [win, rect.shift(delta)];
+const makeTranslate = (delta: RectangleBase) => ([win, rect, offset]: Move): Move => [win, rect.shift(delta), offset];
+function getInitialPositions(win: OpenFinWindow) {
+    return WindowGroups.getGroup(win.groupUuid).map(moveFromOpenFinWindow);
+}
 function handleBoundsChanging(
     win: OpenFinWindow,
     e: any,
-    payloadBounds: RectangleBase,
-    changeType: number
-): Array<[OpenFinWindow, Rectangle]> {
-    let moves: [OpenFinWindow, Rectangle][];
-    const thisRect = createRectangleFromBrowserWindow(win.browserWindow);
-    const newBounds = thisRect.applyOffset(payloadBounds);
-    const initialPositions = WindowGroups
-        .getGroup(win.groupUuid)
-        .map((win) : Move => [win, createRectangleFromBrowserWindow(win.browserWindow)]);
+    rawPayloadBounds: RectangleBase,
+    changeType: number,
+    initialPositions: Move[] = getInitialPositions(win)
+): Move[] {
+    let moves: Move[];
+    const [, start, offset] = moveFromOpenFinWindow(win); //Corrected
+    const end = normalizeExternalBounds(rawPayloadBounds, offset); //Corrected
     switch (changeType) {
```

Comment:
I think it may be time to make each case its own function, this is getting a little to heavy (internal `break`s etc.)